src/javascript/app_2/App/Components/Elements/Popover/popover.jsx

```diff
@@ -1,53 +1,91 @@
-import PropTypes     from 'prop-types';
-import React         from 'react';
-import PopoverBubble from './popover-bubble.jsx';
+import classNames          from 'classnames';
+import PropTypes           from 'prop-types';
+import React               from 'react';
+import { Icon }            from 'Assets/Common/icon.jsx';
+import { IconInfoOutline } from 'Assets/Common/icon-info-outline.jsx';
+import { IconQuestion }    from 'Assets/Common/icon-question.jsx';
+import { IconRedDot }      from 'Assets/Common/icon-red-dot.jsx';
+import PopoverBubble       from './popover-bubble.jsx';
 
 class Popover extends React.PureComponent {
     constructor(props) {
         super(props);
-        this.state = { is_open: false };
-        this.popover_trigger_reference = React.createRef();
+        this.state = {
+            is_open         : false,
+            target_rectangle: null,
+        };
+        this.target_reference = React.createRef();
     }
 
-    onMouseEnter = () => this.setState({ is_open: true });
-    onMouseLeave = () => this.setState({ is_open: false });
+    componentDidMount() {
+        this.setState({
+            is_open         : this.props.has_error,
+            target_rectangle: this.target_reference.current.getBoundingClientRect(),
+        });
+    }
+
+    toggleIsOpen = () => {
+        this.setState({
+            is_open         : !this.state.is_open && Boolean(this.props.message),
+            target_rectangle: this.target_reference.current.getBoundingClientRect(),
+        });
+    }
 
     render() {
         const {
             alignment,
             children,
+            classNameBubble,
```

Comment:
imo, className is enough

Reply:
No, because there is 2 components in `popover` and sometimes you want to target the bubble, sometimes the target icon and sometimes the target.